src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (isCorrectPasswordAndDbContainsUser(password, userFromDbOptional)) {
+            return userFromDbOptional.get();
+        }
```

Comment:
what if optional is empty, is still work? i think you should add: 
```suggestion
        if (userFromDbOptional.isPresent() && isCorrectPasswordAndDbContainsUser(password, userFromDbOptional)) {
            return userFromDbOptional.get();
        }
```